components/mosart/src/cpl/rof_comp_mct.F90

```diff
@@ -582,16 +583,24 @@ subroutine rof_import_mct( x2r_r)
 
     nliq = 0
     nfrz = 0
+    nmud = 0
+    nsan = 0
     do nt = 1,nt_rtm
        if (trim(rtm_tracers(nt)) == 'LIQ') then
           nliq = nt
        endif
        if (trim(rtm_tracers(nt)) == 'ICE') then
           nfrz = nt
        endif
+       if (trim(rtm_tracers(nt)) == 'MUD') then
+          nmud = nt
+       endif
+       if (trim(rtm_tracers(nt)) == 'SAN') then
+          nsan = nt
+       endif
     enddo
-    if (nliq == 0 .or. nfrz == 0) then
-       write(iulog,*) trim(sub),': ERROR in rtm_tracers LIQ ICE ',nliq,nfrz,rtm_tracers
+    if (nliq == 0 .or. nfrz == 0 .or. nmud == 0 .or. nsan == 0) then
```

Comment:
Please split this long if into four ifs.

Reply:
I followed the original style in this line. It was "if (nliq == 0 .or. nfrz == 0) then" for two tracers. I just expanded it since we now have four tracers. I feel we'd better keep it this way.

Reply:
It is not user friendly. When the error message is triggered, the users are difficult to know which tracers have index errors. In addition, instead of a warning message here, an error exception should be used to stop the running of the model.

Reply:
I broke it down into four if statements in v4. "shr_sys_abort" is for stopping the run.